src/Microsoft.AspNetCore.Proxy/ProxyServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿
+using System;
+using Microsoft.AspNetCore.Proxy;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class ProxyServiceCollectionExtensions
+    {
+        public static IServiceCollection AddProxy(this IServiceCollection services)
+        {
+            return services.AddSingleton(new ProxyService(new SharedProxyOptions()));
```

Comment:
Good point.
It is workaroundable by providing singleton instances (for example static ones) when configuring service on per-scope-basis, however I think a lot of people can accidentaly misuse it. What can lead to a disaster which you have pointed out.

However, can you think of any legit scenario for having multiple HttpHandlers?
1. If you need conditional logic you always have DelegatingHandler.
2, WinHttpHandler vs ClientHttpHandler is usually a per-app choice. Legacy version of the ClientHttpHandler (4.0.0 or 4.3.1 on .NET 4.6) is not an option in proxy scenario... because #47 

I really cannot think about any scenario (what doesn't mean there is no scenario).